.github/workflows/python-client-tests.yml

```diff
@@ -0,0 +1,45 @@
+name: Python Client Tests
+
+on:
+  push:
+    branches: [ "main" ]
```

Comment:
i think we need these tests to run on all branches. 

i see that there are other workflows with the same problem. @vikrantpuppala can you fix them?

Reply:
the lines below i.e. 
```
pull_request:
    branches: [ "main" ]
```
will run these tests whenever a PR is created into main. would that not suffice?

Reply:
we are backporting stuff in branch-0.2 with PRs. in those PRs, these tests will not run. that means that we are going to loose test coverage before we release 0.2.x.

Reply:
sure, i would say we could expand the pull request to be main + release branches

Reply:
#624 to track